File: ACE Hardware/ACE Hardware/InventoryEditCategoryEditForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using MySql.Data.MySqlClient;

namespace ACE_Hardware
{
    public partial class InventoryEditCategoryEditForm : Form
    {

        private MySqlConnection conn = new MySqlConnection();
        private string connstring = Program.connstring;
        private string category;

        public InventoryEditCategoryEditForm()
        {
            InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text == "")
            {
                MessageBox.Show("New name must not be blank");
                return;
            }
            if (Program.inventoryEditCategoryFormInstance.ListBox1.Items.Contains(this.TextBox1.Text))
            {
                MessageBox.Show("Category already exists");
                return;
            }
            int categoryindex = 0;
            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "SELECT * FROM productcategorytbl WHERE productcategoryname = '" + category + "'";
                MySqlDataReader reader;
                comm.Connection = conn;
                comm.CommandText = query;
                reader = comm.ExecuteReader();

                while (reader.Read())
                    categoryindex = System.Convert.ToInt32(reader["productcategoryid"].ToString());
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }
            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "UPDATE productcategorytbl SET productcategoryname = '" + TextBox1.Text + "' WHERE productcategoryid = " + categoryindex;

                comm.CommandText = query;
                comm.Connection = conn;
                comm.ExecuteNonQuery();

                MessageBox.Show("Updated");
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }

            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "INSERT INTO logstbl(username,process,description,dateandtime) VALUES('" + Program.login.username + "','Inventory - Edit Category','Edited category (" + category + ")',NOW())";

                comm.Connection = conn;
                comm.CommandText = query;
                comm.ExecuteNonQuery();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }


            this.Close();
            Program.inventoryEditCategoryFormInstance.refreshcategory();
        }

        private void InventoryEditCategoryEditForm_Load(object sender, EventArgs e)
        {
            category = Program.inventoryEditCategoryFormInstance.ListBox1.SelectedItem.ToString();
            Label4.Text = category;
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}

File: POS/POS/InventoryEditCategoryEditForm.vb
Imports MySql.Data.MySqlClient

Public Class InventoryEditCategoryEditForm
    Dim conn As New MySqlConnection
    Dim connstring As String = "Server=localhost;User Id=root;Password=;Database=POS"
    Dim category As String

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If TextBox1.Text = "" Then
            MsgBox("New name must not be blank")
            Exit Sub
        End If
        If InventoryEditCategoryForm.ListBox1.Items.Contains(Me.TextBox1.Text) Then
            MsgBox("Category already exists")
            Exit Sub
        End If
        Dim categoryindex As Integer
        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "SELECT * FROM productcategorytbl WHERE productcategoryname = '" & category & "'"
            Dim reader As MySqlDataReader
            comm.Connection = conn
            comm.CommandText = query
            reader = comm.ExecuteReader

            While reader.Read
                categoryindex = CInt(reader("productcategoryid").ToString)
            End While
        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try
        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "UPDATE productcategorytbl SET productcategoryname = '" & TextBox1.Text & "' WHERE productcategoryid = " & categoryindex

            comm.CommandText = query
            comm.Connection = conn
            comm.ExecuteNonQuery()

            MsgBox("Updated")
        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try

        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "INSERT INTO logstbl(username,process,description,dateandtime) VALUES('" & LoginForm.username & "','Inventory - Edit Category','Edited category (" & category & ")',NOW())"

            comm.Connection = conn
            comm.CommandText = query
            comm.ExecuteNonQuery()
        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try


        Me.Close()
        InventoryEditCategoryForm.refreshcategory()
    End Sub

    Private Sub InventoryEditCategoryEditForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        category = InventoryEditCategoryForm.ListBox1.SelectedItem.ToString()
        Label4.Text = category
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me.Close()
    End Sub
End Class